docs/what-it-does.md

```diff
@@ -0,0 +1,103 @@
+# What the Consul Deployment Agent Does
+
+## On startup
+
+- Waits for puppet to drop file to signal instance readiness before proceeding.
+
+- Loads consul and standard configuration from the local config.yml file.
+
+- Loads server role information from the host EC2 instance's tags (or hard-coded values during local development).
+
+- Figures out server role key and constantly polls (long polling) for changes in key value store. When a change is detected the convergence process is run (see [Convergence](#convergence) process below). Key format is... `environments/{ENVIRONMENT}/roles/{SERVER_ROLE}`
+
+<a id="convergence"></a>
+
+## Convergence
+
+- Loads list of serviceSlices registered in the catalog.
+
+- Loads list of serviceSlices in KV store.
+
+	- Key to list services: `environments/{ENVIRONMENT}/roles/{SERVER_ROLE}/services`
+	- Key to list slices for a service: `environments/{ENVIRONMENT}/roles/{SERVER_ROLE}/services/{SERVICE_NAME}`
+
+- For each serviceSlice listed...
+
+  - pulls details from: `environments/{ENVIRONMENT}/roles/{SERVER_ROLE}/services/{SERVICE_NAME}/{SLICE}`. This gives us the deploymentId and version number. We then collect further details from:
+    - `environments/{ENVIRONMENT}/services/{SERVICE_NAME}/{VERSION}/definition`
+    - `environments/{ENVIRONMENT}/services/{SERVICE_NAME}/{VERSION}/installation`
+    - `deployments/{DEPLOYMENT_ID}`
+  - Figures out which serviceSlices are not present or have the wrong version.
+  - Filters serviceSlices which are in 'Ignore' state or which have beenpreviously quarantened.
+  - Deploys each serviceSlice in sequence (see [Deployment](#deployment) below).
+
+<a id="deployment"></a>
+
+## Deployment
+
+- Creates a new log / report for the deployment
+
+- Update the status of the node (started deployment) in the KV store.Key: deployments/{DEPLOYMENT_ID}/nodes/{NODE_ID}
+
+- Runs the deployment stages in sequence (see [Deployment Stages](#deployment-stages) for more details)
+	
+- If any of the stages above fail then report failed and attempt to run stop stage, otherwise report success
+
+- Finalise the log and the report
+
+- Update the status of the node (finished deployment) in the KV store. Key: `deployments/{DEPLOYMENT_ID}/nodes/{NODE_ID}`
+
+<a id="deployment-stages"></a>
+
+## Deployment Stages
+
+- Of the above stages the following need to be provided in the deployment package...
+
+	- StopApplication
+	- BeforeInstall
+	- AfterInstall
+	- StartApplication
+	- ValidateService
+
+- The location of the script files in the package are specified by the appspec.yml.
+
+- They are executed in the above order.
```

Comment:
As far as I am aware, you can provice the script location, the timeout for that script and the runas user. Is it worth highlighting what you can add to this yml file?